cooltools/api/coverage.py

```diff
@@ -102,24 +125,40 @@ def coverage(
         raise ValueError(
             "Please, specify ignore_diags and/or IC balance this cooler! Cannot access the value used in IC balancing. "
         )
-
+    if clr_weight_name is not None:
+        try:
+            _ = is_cooler_balanced(clr, clr_weight_name, raise_errors=True)
+        except Exception as e:
+            raise ValueError(
+                f"provided cooler is not balanced or {clr_weight_name} is missing"
+            ) from e
+            
     chunks = cooler.tools.split(clr, chunksize=chunksize, map=map, use_lock=use_lock)
 
     if ignore_diags:
         chunks = chunks.pipe(_zero_diags, n_diags=ignore_diags)
 
     n_bins = clr.info["nbins"]
-    covs = chunks.pipe(_get_chunk_coverage).reduce(np.add, np.zeros((2, n_bins)))
-    covs = covs[0].astype(int), covs[1].astype(int)
+    get_chunk_coverage = partial(_get_chunk_coverage, clr_weight_name=clr_weight_name)
+    covs = chunks.pipe(get_chunk_coverage).reduce(np.add, np.zeros((2, n_bins)))
+    if clr_weight_name is None:
+        covs = covs[0].astype(int), covs[1].astype(int)
+    elif store_names == ["cis_raw_cov", "tot_raw_cov"]:
+        store_names = [s + str("_"+clr_weight_name) for s in store_names]
 
     if store:
         with clr.open("r+") as grp:
+            if clr_weight_name is None:
```

Comment:
IMHO this is one of the rare occasions where a one-liner `if-else` works better:
```python
dtype = int if clr_weight_name is None else float
```

but that's just taste ...